src/lib/client/jubSignalClient.ts

```diff
@@ -0,0 +1,296 @@
+          if (user) {
+            user.name = name;
+            user.encPk = pk;
+            user.x = user.x && x === undefined ? user.x : x;
+            user.tg = user.tg && tg === undefined ? user.tg : tg;
+            user.note = note;
+            user.outTs = metadata.timestamp.toISOString();
+
+            users[userId] = user;
+          } else {
+            users[userId] = {
+              name,
+              encPk: pk,
+              x,
+              tg,
+              note,
+              outTs: metadata.timestamp.toISOString(),
+            };
+          }
+
+          const activity = {
+            type: JUB_SIGNAL_MESSAGE_TYPE.OUTBOUND_TAP,
+            name,
+            id: userId,
+            ts: metadata.timestamp.toISOString(),
+          };
+          activities.push(activity);
+        } catch (error) {
+          console.error(
+            "Invalid outbound tap message received from server: ",
+            message
+          );
+        } finally {
+          break;
+        }
+      case JUB_SIGNAL_MESSAGE_TYPE.INBOUND_TAP:
+        // TODO: Can optionally validate received signature here
+        try {
+          const { x, tg, pk, msg, sig } =
+            await inboundTapMessageSchema.validate(data);
+          const userId = await hashPublicKeyToUUID(metadata.fromPublicKey);
+          const user = users[userId];
+          if (user) {
+            user.name = metadata.fromDisplayName;
+            user.encPk = metadata.fromPublicKey;
+            user.x = x;
+            user.tg = tg;
+            user.sigPk = pk;
+            user.msg = msg;
+            user.sig = sig;
+            user.inTs = metadata.timestamp.toISOString();
+
+            users[userId] = user;
+          } else {
+            users[userId] = {
+              name: metadata.fromDisplayName,
+              encPk: metadata.fromPublicKey,
+              x,
+              tg,
+              sigPk: pk,
+              msg,
+              sig,
+              inTs: metadata.timestamp.toISOString(),
+            };
+          }
+
+          const activity = {
+            type: JUB_SIGNAL_MESSAGE_TYPE.INBOUND_TAP,
+            name: metadata.fromDisplayName,
+            id: userId,
+            ts: metadata.timestamp.toISOString(),
+          };
+          activities.push(activity);
+        } catch (error) {
+          console.error(
+            "Invalid inbound tap message received from server: ",
+            message
+          );
+        } finally {
+          break;
+        }
+      case JUB_SIGNAL_MESSAGE_TYPE.LOCATION_TAP:
+        try {
+          if (metadata.fromPublicKey !== recipientPublicKey) {
+            throw new Error(
+              "Invalid message: location tap messages must be sent from self"
+            );
+          }
+
+          const { id, name, pk, msg, sig } =
+            await locationTapMessageSchema.validate(data);
+          const location = locationSignatures[id];
+          if (location) {
```

Comment:
Same comment as above: Why don't we just replace locationSignatures[id] fully? Like what is the difference between the if and else here?